packages/opal-client/opal_client/data/updater.py

```diff
@@ -327,109 +330,38 @@ async def update_policy_data(
-                            url=url,
-                            status=result.status,
-                            error=error_content,
-                        )
-                    except json.JSONDecodeError:
-                        error_content = await result.text()
-                        logger.error(
-                            "Failed to decode response from url:{url}, got response code {status} with response: {error}",
-                            url=url,
-                            status=result.status,
-                            error=error_content,
-                        )
-                store_transaction._update_remote_status(
-                    url=url, status=fetched_data_successfully, error=str(error_content)
-                )
-
-                if fetched_data_successfully:
-                    # get path to store the URL data (default mode (None) is as "" - i.e. as all the data at root)
-                    policy_store_path = "" if entry is None else entry.dst_path
-                    # None is not valid - use "" (protect from missconfig)
-                    if policy_store_path is None:
-                        policy_store_path = ""
-                    # fix opa_path (if not empty must start with "/" to be nested under data)
-                    if policy_store_path != "" and not policy_store_path.startswith(
-                        "/"
-                    ):
-                        policy_store_path = f"/{policy_store_path}"
-                    policy_data = result
-                    # Create a report on the data-fetching
-                    report = DataEntryReport(
-                        entry=entry, hash=self.calc_hash(policy_data), fetched=True
-                    )
-                    logger.info(
-                        "Saving fetched data to policy-store: source url='{url}', destination path='{path}'",
-                        url=url,
-                        path=policy_store_path or "/",
-                    )
-                    try:
-                        await store_transaction.set_policy_data(
-                            policy_data, path=policy_store_path
-                        )
-                        # No exception we we're able to save to the policy-store
-                        report.saved = True
-                        # save the report for the entry
-                        reports.append(report)
-                    except Exception:
-                        logger.exception("Failed to save data update to policy-store")
-                        # we failed to save to policy-store
-                        report.saved = False
-                        # save the report for the entry
-                        reports.append(report)
-                        # re-raise so the context manager will be aware of the failure
-                        raise
+                if entry.save_method.upper() == "DELETE":
```

Comment:
Again, I don't think I like the special handling of "DELETE". 
I don't think it makes sense to split it to different methods here - `PolicyStoreClient` might need to have different implementations for different methods but that's not the concern of this part.

I know this method have become too big and messy over time, but we should fix it in some other way (or possibly in another PR, I thought about separating all the fetching & http status codes handling to another method called `enrich_entries` which fetches urls, and could do other stuff in the future).

Moreover, as a result of your separation of `handle_delete_data` from `handle_new_data`:
1. path normalization isn't happening for "DELETE".
2. reports are not collected for "DELETE".
3. reports collected for other entries are not reported.

**In short - I think you should basically let this code handle DELETE entries like any other entry** (maybe should make `DataFetcher.handle_url` not log an error when both url & data are None, or if you have the time implement that `enrich_entries` which proxies `handle_urls` but not letting through entries of that kind)